parsec/core/fs/workspacefs/sync_transactions.py

```diff
@@ -52,19 +52,25 @@ def get_conflict_filename(
     return new_filename
 
 
-def full_name(name: EntryName, *suffixes: str) -> EntryName:
-    # No suffix
-    if not suffixes:
-        return name
-
-    # No extension
-    suffix_string = "".join(f" ({suffix})" for suffix in suffixes)
-    if "." not in name[1:]:
-        return EntryName(name + suffix_string)
-
-    # Extension
-    first_name, *ext = name.split(".")
-    return EntryName(".".join([first_name + suffix_string, *ext]))
+def full_name(name: EntryName, suffix: str) -> EntryName:
+    # Format the suffix string
+    suffix_string = f" ({suffix})"
+    # Separate file name from the extentions (if any)
+    name_parts = name.split(".")
+    non_empty_indexes = (i for i, part in enumerate(name_parts) if part)
+    first_non_empty_index = next(non_empty_indexes, len(name_parts) - 1)
+    base_name = ".".join(name_parts[: first_non_empty_index + 1])
+    extensions = name_parts[first_non_empty_index + 1 :]
+    # Loop over attemps, in case the produced entry name is too long
+    while True:
+        # Convert to EntryName
+        try:
+            return EntryName(".".join([base_name + suffix_string, *extensions]))
+        # Entry name too long
+        except EntryNameTooLongError:
+            # Simply strip 10 characters from the first name then try again
+            assert len(base_name) > 0
```

Comment:
we can reach this assert if extensions are too long, for instance `python -c 'print("foo." + "a" * 251)'`  generates a valid name, but which will trigger this assert if the file need to be conflict renamed

I think it's "good enough" right now, but should be taken into account in the next PR that replace the device name by a simpler rename scheme

Reply:
If we reach the assert that means there's no room in the base name to add the suffix, so I'm not sure about what should be done in that case.

Reply:
65c8edbf94a88609907f7dbec4c55e7dd9c31ca3